pandas/io/sql.py

```diff
@@ -1453,15 +1453,22 @@ def insert_statement(self):
 
         bracketed_names = [escape(column) for column in names]
         col_names = ",".join(bracketed_names)
-        wildcards = ",".join([wld] * len(names))
+
+        row_wildcards = f'({",".join([wld] * len(names))})'
+        wildcards = ",".join(row_wildcards for _ in range(num_rows))
         insert_statement = (
-            f"INSERT INTO {escape(self.name)} ({col_names}) VALUES ({wildcards})"
+            f"INSERT INTO {escape(self.name)} ({col_names}) VALUES {wildcards}"
         )
         return insert_statement
 
     def _execute_insert(self, conn, keys, data_iter):
         data_list = list(data_iter)
-        conn.executemany(self.insert_statement(), data_list)
+        conn.executemany(self.insert_statement(num_rows=1), data_list)
+
+    def _execute_insert_multi(self, conn, keys, data_iter):
+        data_list = list(data_iter)
+        flattened_data = [x for row in data_list for x in row]
```

Comment:
Two copies as this ended up being the clearest way of getting both the number of rows and the rows flattened correctly.

In principle we can avoid doing that with something like:

```python
flattened_data = []
num_rows = 0
for row in data_iter:
    flattened_data.extend(row)
    num_rows += 1
```

but I don't see a compelling performance reason to take the complexity hit.